### ISSUE TITLE
[Discussion] UbuntuWRT Releases

### ISSUE THREAD

**Issue by cilix-lab** [cilix-lab/ubuntu-wrt#3]
From now on, updates and related discussion will be posted in this thread.  

**UbuntuWRT 17.10 has been released, with several improvements and upgraded base system to Ubuntu 17.04 Zesty!**  

Check out the README.md to learn more.

**Comment by cilix-lab** [comment]
UbuntuWRT 17.10.1 has been released.  

Changelog:  
* Updated kernel to 4.10.17-40.44-0.  
* Updated mwlwifi to commit 466368f.  
* Changed network configuration to properly set eth1 MAC from script.  
* Minor fixes to some configuration files.  

Checkout [README.md](https://github.com/cilix-lab/ubuntu-wrt/blob/master/README.md) for download links.

**Comment by cilix-lab** [comment]
**mwlwifi update to commit [1522af5](https://github.com/kaloz/mwlwifi/commit/1522af59a501f74c8b6b02ae763829a79114a325).**  

**linux-modules** package has been renamed to **linux-image** and sources are now available too as linux-headers.  

To update:  
```
# Update modules
apt-get update
apt-get remove linux-modules
apt-get install linux-image

# Router will be flashed and ask for a reboot.
# If anything fails in dpkg configuration step, you can manually
# flash the file in /lib/firmware/wrt

# Get kernel source (optional)
apt-get install linux-headers
```

**Comment by cilix-lab** [comment]
**mwlwifi update to commit [af210f0](https://github.com/kaloz/mwlwifi/commit/af210f0ef79d5c416dd455747a16d76bec5912ba).**  

**Firmware update to 0x09030201.**  

CRDA support in the kernel has been disabled due to issues with the new firmware.  

To update, if you're on linux-image 4.10.17-40.44-1, just: `apt-get update && apt-get upgrade`

**Comment by cilix-lab** [comment]
**mwlwifi update to commit [843d00c](https://github.com/kaloz/mwlwifi/commit/843d00cd9c134629b9dad7162831ec5f136399b3).**  

**Firmware update to 0x09030202.** Fixed issue where firmware region code wouldn't be set.  

CRDA support has been enabled again. Make sure your `/etc/default/crda` file is set properly.  

Ubuntu kernel has been updated to 4.10.17-42.46.  

If you're in the latest rootfs, just: `apt-get update && apt-get upgrade`

**Comment by cilix-lab** [comment]
**UbuntuWRT 18.01-beta is here!**  

Checkout the README.md.  

Changelog:
* Reverted back to 16.04 Xenial for LTS.
* Implemented DSA switch.
* Fixed interface configurations. Now booting takes only seconds.
* Back to mainline kernel. Current version is 4.14.13.
* Implemented ValCher1961's region free patch, so you can now set your region properly.

**Comment by Chadster766** [comment]
Well done :smiley:

**Comment by cilix-lab** [comment]
Never took the "beta" out of "UbuntuWRT 18.01-beta" and yet official Ubuntu 18.04 is almost here, so I have some good news about it.  

I am currently working on moving forward to UbuntuWRT 18.04, based on 18.04 Bionic Beaver, and implementing some big changes:  

* Removing BIND9 as DNS server.
* Adding Samba as Active Directory Domain Controller and using it's built in DNS server. That will enable you to deploy a Windows Domain network and have a centralized secure location for user management and services or... if you have no idea what all of that is, you'll get a pretty name when connecting Windows clients to UbuntuWRT's network instead of the usual "Network #".
* Dynamic DNS updates from ISC-DHCP-Server to Samba DNS for DHCP clients.
* [DNScrypt-Proxy](https://dnscrypt.info) for secure DNS resolving. Your ISP might pry on your traffic, but at least they won't see your DNS queries.

Roadmap:  

* Develop web GUI for initial setup.
* Develop web GUI for statistics.

Let me know what you think. Mostly about the move from BIND9 DNS to Samba AD DC and Samba DNS.

**Comment by Redferne** [comment]
Personally I do not give two cents for Windows clients or SMB. I prefer dnsmasq and NFS. Please keep the bloating to minimum since we only have 512MB to play with.

**Comment by tienhuynh5312** [comment]
Just a clarification. Will DNScrypt-Proxy be on 2.0?
Thank you. Keep up good work!

**Comment by cilix-lab** [comment]
OK. giving it some though, I realized that most people are probably on @Redferne side and don't really need fancy setups. So, over the weekend I got Ubuntu 18.04 Bionic Beta working on the WRT3200ACM, with a simple dnsmasq setup.  

I'm keeping DNScrypt-Proxy, though, v2.0 @tienhuynh5312, which I've packaged for UbuntuWRT and is currently available through the repo (with version 2.0.9-0wrt2).  

I'll make a script available for setting up an Active Directory Domain Controller if needed, which should replace dnsmasq with Samba DNS and ISC-DHCP-Sever.  

I'll update the repo and release the firmware and rootfs when I'm done with some tests.  

Thanks for the feedback!

**Comment by cilix-lab** [comment]
**UbuntuWRT 18.04** is here!  

Check the updated README.md for more info and links for the ROOTFS and firmware.  

If you are on 18.01 with latest linux-image installed, then you don't need to reflash your router, giving you the advantage of preparing the rootfs on a thumb drive, try it on the router without flashing and being able to switch thumb drives again to go back to your previous config.
